xds/cmd/controller.go

```diff
@@ -70,18 +97,53 @@ func createAgonesClusterProvider(
 		externalversions.WithNamespace(flags.GameServersNamespace))
 	informerFactory.Start(ctx.Done())
 
+	gameServerInformer := informerFactory.Agones().V1().GameServers().Informer()
+	err = gameServerInformer.SetWatchErrorHandler(func(r *cache.Reflector, err error) {
```

Comment:
I've not seen `SetWatchErrorHandler` before!

That makes sense, given you aren't writing back to the K8s control plane, you can't pick up disconnections otherwise 👍🏻

Question: Wouldn't the health check ping on the k8s control plane capture this, without the extra handler? (I mean, you get more logs this way, but just thinking about it, once you know it's disconnected, it's because the whole thing has gone away - it would be very weird if you disconnected from just GameServers and not Pods for example -- it's the same backend).

Reply:
removed